lrspline/__init__.py

```diff
@@ -461,18 +479,32 @@ def derivative(self, u, v, d=(1,1), iel=-1):
             return np.array(retval).reshape(u.shape)
         return self.w.point(u, v, nderivs, iEl=iel)[index]
 
+    def getBezierExtraction(self, iEl):
+        return self.w.getBezierExtraction(iEl)
+
     __call__ = evaluate
 
 
 class LRSplineVolume(LRSplineObject):
 
-    def __init__(self, arg=None):
-        if isinstance(arg, raw.LRVolume):
+    def __init__(self, *args):
+        if len(args) == 0:
+            w = raw.LRVolume()
+        elif isinstance(args[0], raw.LRVolume):
             w = arg
+        elif isinstance(args[0], io.IOBase):
+            w = raw.LRVolume()
+            w.read(args[0])
+        elif len(args) == 3: # only specify number of functions (n1,n2,n3)
+            w = raw.LRVolume(args[0], args[1], args[2], 2, 2, 2)
+        elif len(args) == 6: # specity n & p for 3 directions
```

Comment:
Same here, these three paths are the same, so you can probably just do `if 6 <= len(args) <= 10`.

Reply:
The problem is when stupid users gives 5 input arguments. I decided that the best way was to just return the empty constructor in those cases. The reason for splitting it into multiple lines was for readability. While you can get away with:

``` python
if len(args) in (3,6,9,10): w = raw.LRVolume(*args)
```
is a one-line solution, I actually prefer a little longer more verbose implementation for documentation purposes.